Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -400,6 +438,76 @@ private static bool ShouldRequirePassword(Contest contest, Participant participa
             return (official && contest.HasContestPassword) || (!official && contest.HasPracticePassword);
         }
 
+        private async Task<PagedResult<ContestForListingServiceModel>> PrepareActivityAndResults(
+            PagedResult<ContestForListingServiceModel> pagedContests,
+            Dictionary<int, List<ParticipantResultServiceModel>>? participantResultsByContest = null)
+        {
+            var participantsCount = await this.contestParticipantsCacheService
+                .GetParticipantsCount(
+                    pagedContests.Items
+                        .Select(c => c.Id)
+                        .Distinct()
+                        .ToImmutableArray(),
+                    pagedContests.PageNumber);
+
+            pagedContests.Items.ForEach(c =>
+            {
+                this.activityService.SetCanBeCompetedAndPracticed(c);
+                c.CompeteResults = participantsCount[c.Id].Official;
+                c.PracticeResults = participantsCount[c.Id].Practice;
+
+                if (participantResultsByContest != null)
+                {
+                    if (participantResultsByContest[c.Id].Count > 1)
+                    {
+                        // Compete and practice participants
+                        var practiceParticipant = participantResultsByContest[c.Id].First(p => !p.IsOfficial);
+
+                        var competeParticipant = participantResultsByContest[c.Id].First(p => p.IsOfficial);
```

Comment:
Are we sure that there will always be entry here (using `First`)? Contest will not always have both official and unofficial participants

Reply:
Yes, works okay:
![image](https://github.com/SoftUni-Internal/OpenJudgeSystem/assets/19411274/9291d155-b694-4bf0-87bf-e494859aeb1a)

Reply:
Now I see that it's in an `if` check for participants count, that's why it's passing.
Looking at the logic, I see a simpler way of doing it, without extra `if` statement:
```
var participants = participantResultsByContest[c.Id];
var competeParticipantPoints = participants.Where(p => p.IsOfficial).Select(p => p.Points).SingleOrDefault();
var practiceParticipantPoints = participants.Where(p => !p.IsOfficial).Select(p => p.Points).SingleOrDefault();

c.UserParticipationResult = new ContestParticipantResultServiceModel
{
    CompetePoints = competeParticipantPoints,
    PracticePoints = practiceParticipantPoints,
};
```